### ISSUE TITLE
PyPy uses wrong file name encoding on Windows

### ISSUE THREAD

**Issue by agronholm** [pypy/pypy#4924]
The following script demonstrates 2 seemingly different issues:
```python
from pathlib import Path

path = Path("MARKER-ö.txt")
path.touch()

path2 = next(p for p in Path.cwd().iterdir() if p.name.startswith("MARKER-"))
path2.chmod(0o777)
```

1. The created file is named `MARKER-Ã¶.txt` and not `MARKER-ö.txt` as is should be (and is on CPython)
2. The chmod operation fails with `FileNotFoundError`, despite just being returned from `iterdir()`, likely also due to a file name encoding mismatch

The issue appears to be caused by some code erroneously using the iso-latin-1 encoding, as the UTF-8 bytes 0xC3 0xB6 decoded as latin-1 translate exactly to `Ã¶`.

**Comment by mattip** [comment]
Related to #3890. Is this with latest PyPy 7.3.15?

**Comment by agronholm** [comment]
Yes, I just downloaded it today.

**Comment by mattip** [comment]
When I try with 7.3.15 in a cmd.exe terminal, I get
```
>chcp
Active code page: 437
>pypy3.9-v7.3.15-win64\python.exe
Python 3.9.18 (9c4f8ef178b6, Jan 14 2024, 12:40:23)
[PyPy 7.3.15 with MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>> import pathlib
>>>> path = Path("MARKER-ö.txt")
UnicodeEncodeError: 'utf-8' codec can't encode character '\udc94' in position 20: surrogates not allowed
```

So that is one problem. To get your filename, I can do
```
from pathlib import Path
name = b'MARKER-\xc3\xb6.txt'.decode()
ord(name[7])
246  # that checks out with `ord('ö')`
path = Path(name)
path.touch()
```

Now I see a file with the name `MARKER-Ã¶.txt` in FileExplorer, just like in the issue report.

**Comment by agronholm** [comment]
For me, `chcp` reports 850 as the active code page.

**Comment by mattip** [comment]
The path to solving this goes via adding a test that can be run untranslated, trying to figure out where the encoding goes bad, and then fixing it.

**Comment by mattip** [comment]
The reproducer now works correctly on py3.10 HEAD. Closing. Please reopen if I missed something.

**Comment by kurtmckee** [comment]
@mattip

> The reproducer now works correctly on py3.10 HEAD

I'm seeing something that looks related to this in PyPy 3.9 on Windows (I'm testing this in CI, so I haven't explored this on a local Windows box). It works on PyPy 3.10, just as you noted above:

```python
# Given: a file named `dir/あ.py` already exists
import pathlib

directory = pathlib.Path("dir")
file = pathlib.Path("あ.py")

(directory / file).read_bytes()  # FileNotFound
```

I tried working around this by encoding the filename to bytes using UTF-8, but while CI passed for Linux and macOS, PyPy 3.9 AND PyPy 3.10 failed on Windows:

```python
# Given: the same `directory` and `file` as above
...

filename = str(directory / file).encode("utf-8")
with open(filename, "rb") as f:  # FileNotFound
    raw_contents = f.read()
```

If this should be a different issue, let me know, otherwise I'm pinging here because this issue does not yet appear to be resolved for PyPy 3.9.

**Comment by kurtmckee** [comment]
Oh! I didn't realize that PyPy 3.9 was no longer supported. I'll stop testing against that version.

The output from CI is:

[actions/setup-python output](https://github.com/kurtmckee/sqliteimport/actions/runs/12121470165/job/33792463200#step:12:21):

```
Successfully set up PyPy 7.3.17 with Python (3.10.14)
```

[pytest output just prior to running the test suite](https://github.com/kurtmckee/sqliteimport/actions/runs/12121470165/job/33792463200#step:18:154):

```
platform win32 -- Python 3.10.14[pypy-7.3.17-final], pytest-8.3.3, pluggy-1.5.0
```

And finally, here's [a direct link to the `FileNotFound` error](https://github.com/kurtmckee/sqliteimport/actions/runs/12121470165/job/33792463200#step:18:202).
